src/main/java/org/minutebots/frc2019/Robot.java

```diff
@@ -9,8 +28,51 @@
    */
   @Override
   public void robotInit() {
+      
+  }
+
+  final Encoder leftEncoder = new Encoder(0, 0);
+  final Encoder rightEncoder = new Encoder(0, 0);
+  final DifferentialDriveOdometry m_odometry = new DifferentialDriveOdometry(new Rotation2d(0.0));
+
+  private Ramsete generateRamsete(){
+
+    final Pose2d start = new Pose2d();
+    final Pose2d end = new Pose2d();
+    final TrajectoryConfig  config = new TrajectoryConfig(1.0, 1.0);
+    final ArrayList<Translation2d> interiorWaypoints = new ArrayList<Translation2d>();
+    final SimpleMotorFeedforward feedforward = new SimpleMotorFeedforward(0,0);
+    final DifferentialDriveKinematics kinematics = new DifferentialDriveKinematics(0);
+
+    final PIDController leftController = new PIDController(0,0,0);
+    final PIDController rightController = new PIDController(0,0,0);
+
+
+      final Trajectory trajectory = TrajectoryGenerator.generateTrajectory(start, interiorWaypoints, end, config);
+    
+      final RamseteController controller = new RamseteController();
+    
+      return new Ramsete(trajectory, getPose, controller, feedforward, kinematics, this::wheelSpeeds, leftController, rightController, this::huh);
+  }
+
+  private Supplier<Pose2d> getPose = new Supplier<Pose2d>(){
+
+    public Pose2d get() {
+      return m_odometry.update(new Rotation2d(gyroAngle()), leftEncoder.getRate(), rightEncoder.getRate());
+    }
+
+  };
+
+  private DifferentialDriveWheelSpeeds wheelSpeeds(){
+    return new DifferentialDriveWheelSpeeds(leftEncoder.getRate(),rightEncoder.getRate());
+  }
+  private void huh(Double a, Double b){
+    
   }
 
+  private double gyroAngle(){
+    return 0;
+  }
```

Comment:
It would make sense to rewrite this assuming that we do have a functioning gyro. A check to see if a gyro exists then returning 0 if false would suffice.